File: ugs-pendant/src/main/webapp/src/app/model/macro.ts
export class Macro {
  private _name: string;
  private _description: string;
  private _gcode: string;

  constructor(name:string = '', description:string = '', gcode:string = '') {
    this._name = name;
    this._description = description;
    this._gcode = gcode;
  }

  get name(): string {
    return this._name;
  }

  set name(name:string) {
    this._name = name;
  }

  get description():string {
    return this._description;
  }

  set description(description:string) {
    this._description = description;
  }

  get gcode(): string {
    return this._gcode;
  }

  set gcode(gcode:string) {
    this._gcode = gcode;
  }
}

File: ugs-pendant/src/main/java/com/willwinder/universalgcodesender/pendantui/v1/model/Macro.java
package com.willwinder.universalgcodesender.pendantui.v1.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Macro implements Serializable {
    // Was missing entirely before - fine for a read-only "list and run"
    // view, but update/delete need a stable identifier to target, and the
    // list order (used for reordering) needs something to key React list
    // items on that survives a rename.
    private String uuid;
    private String gcode;
    private String description;
    private String name;
    private String color;
    private String icon;

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getGcode() {
        return gcode;
    }

    public void setGcode(String gcode) {
        this.gcode = gcode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }
}
